In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [23]:
path = '/content/drive/MyDrive/mulcam_bigdata/2024통계데이터공모전/기획_데이터수집/성원님/temp/MCLP_Data/'

# 1차소아과병원 시도시군구 추출
- 시도별코드 매핑해주기 위함

In [24]:
입지후보지 = pd.read_csv(path+'전국1차소아과위치전처리.csv')
입지후보지

요양기관명    중분류  시군구       시도  \
0         (사)인구보건복지협회 부산시지회 가족보건의원  21140  수영구    부산광역시   
1        (사)인구보건복지협회 충북세종지회 가족보건의원  33040  청주시     충청북도   
2             (사단법인)한국건강관리협회건강증진의원  24020   서구    광주광역시   
3               (사회복지법인)한국혈우재단광주의원  24020   서구    광주광역시   
4     (재)한국산업보건연구재단 서울DMC건강의원 포항분원  37010  포항시     경상북도   
...                            ...    ...  ...      ...   
2607                        강북구보건소  11090  강북구    서울특별시   
2608                    대구광역시서구보건소  22030   서구    대구광역시   
2609                        은평구보건소  11120  은평구    서울특별시   
2610                    군산시어청도보건지소      0  군산시  전북특별자치도   
2611                    완도군노화읍보건지소  36460  완도군     전라남도   

                                              주소         위도          경도  
0                       부산광역시 수영구 수영로 425, (남천동)  35.144097  129.108936  
1            충청북도 청주시 흥덕구 내수동로 20, (복대동, 모자보건센타)  36.632997  127.449528  
2     광주광역시 서구 대남대로 432, 한국건강관리협회 지하2~지상7층 (농성동)  35.149037  126.888342  
3                     광주광역시 서구 죽봉대로 66, 4층 (농성동)  35.159100  126.883799  
4          경상북도 포항시 남구 오천읍 중원로 71-11, 1동 1층 101호  35.975330  129.400240  
...                                          ...        ...         ...  
2607      서울특별시 강북구 한천로 897, (번동, 강북보건소,번2동주민센터)  37.632333  127.038767  
2608                   대구광역시 서구 국채보상로 257, (평리동)  35.871815  128.559177  
2609                    서울특별시 은평구 은평로 195, (녹번동)  37.602671  126.928877  
2610              전북특별자치도 군산시 옥도면 어청도길 98, (옥도면)  36.120882  125.980074  
2611                전라남도 완도군 노화읍 노화로 813, 노화보건지소  34.177240  126.571151  

[2612 rows x 7 columns]

In [27]:
입지후보지[입지후보지['중분류'] == 0]

요양기관명  중분류  시군구       시도  \
8     21세기소아청소년과의원    0  원주시  강원특별자치도   
48        강남제이스타의원    0  원주시  강원특별자치도   
72      강훈소아청소년과의원    0  원주시  강원특별자치도   
75      개성소아청소년과의원    0  속초시  강원특별자치도   
97     고양심소아청소년과의원    0  완주군  전북특별자치도   
...            ...  ...  ...      ...   
2510    햇님소아청소년과의원    0  전주시  전북특별자치도   
2523   행복한소아청소년과의원    0  강릉시  강원특별자치도   
2564       호산부인과의원    0  춘천시  강원특별자치도   
2587   황선미소아청소년과의원    0  전주시  전북특별자치도   
2610    군산시어청도보건지소    0  군산시  전북특별자치도   

                                                     주소         위도          경도  
8                             강원특별자치도 원주시 봉화로 27, (단계동)  37.346919  127.928258  
48                       강원특별자치도 원주시 서원대로 180, 4층 (단계동)  37.343837  127.931309  
72        강원특별자치도 원주시 입춘로 110, (반곡동, 파라다이스프라자 301,302호)  37.323406  127.981684  
75                          강원특별자치도 속초시 동해대로 4215, (교동)  38.195245  128.577224  
97                         전북특별자치도 완주군 봉동읍 봉동로 135, 135  35.937090  127.168108  
...                                                 ...        ...         ...  
2510  전북특별자치도 전주시 덕진구 쪽구름로 74, 전라북도 전주시 덕진구 쪽구름로 74,...  35.871956  127.076487  
2523                   강원특별자치도 강릉시 성덕포남로182번길 46, (포남동)  37.770037  128.913358  
2564                         강원특별자치도 춘천시 퇴계로 189, (석사동)  37.857516  127.739807  
2587  전북특별자치도 전주시 완산구 효천중앙로 22, A동 1층 105호 일부&106호 전...  35.797507  127.104407  
2610                     전북특별자치도 군산시 옥도면 어청도길 98, (옥도면)  36.120882  125.980074  

[170 rows x 7 columns]

In [29]:
# 주소 데이터를 시도와 시군구로 구분

입지후보지['시도'] = 입지후보지['주소'].apply(lambda x: x.split(' ', 1)[0])
입지후보지['시군구'] = 입지후보지['주소'].apply(lambda x: x.split(' ', 2)[1])
입지후보지['시군구2'] = 입지후보지['주소'].apply(lambda x: x.split(' ', 3)[2])

In [30]:
입지후보지

요양기관명    중분류  시군구       시도  \
0         (사)인구보건복지협회 부산시지회 가족보건의원  21140  수영구    부산광역시   
1        (사)인구보건복지협회 충북세종지회 가족보건의원  33040  청주시     충청북도   
2             (사단법인)한국건강관리협회건강증진의원  24020   서구    광주광역시   
3               (사회복지법인)한국혈우재단광주의원  24020   서구    광주광역시   
4     (재)한국산업보건연구재단 서울DMC건강의원 포항분원  37010  포항시     경상북도   
...                            ...    ...  ...      ...   
2607                        강북구보건소  11090  강북구    서울특별시   
2608                    대구광역시서구보건소  22030   서구    대구광역시   
2609                        은평구보건소  11120  은평구    서울특별시   
2610                    군산시어청도보건지소      0  군산시  전북특별자치도   
2611                    완도군노화읍보건지소  36460  완도군     전라남도   

                                              주소         위도          경도   시군구2  
0                       부산광역시 수영구 수영로 425, (남천동)  35.144097  129.108936    수영로  
1            충청북도 청주시 흥덕구 내수동로 20, (복대동, 모자보건센타)  36.632997  127.449528    흥덕구  
2     광주광역시 서구 대남대로 432, 한국건강관리협회 지하2~지상7층 (농성동)  35.149037  126.888342   대남대로  
3                     광주광역시 서구 죽봉대로 66, 4층 (농성동)  35.159100  126.883799   죽봉대로  
4          경상북도 포항시 남구 오천읍 중원로 71-11, 1동 1층 101호  35.975330  129.400240     남구  
...                                          ...        ...         ...    ...  
2607      서울특별시 강북구 한천로 897, (번동, 강북보건소,번2동주민센터)  37.632333  127.038767    한천로  
2608                   대구광역시 서구 국채보상로 257, (평리동)  35.871815  128.559177  국채보상로  
2609                    서울특별시 은평구 은평로 195, (녹번동)  37.602671  126.928877    은평로  
2610              전북특별자치도 군산시 옥도면 어청도길 98, (옥도면)  36.120882  125.980074    옥도면  
2611                전라남도 완도군 노화읍 노화로 813, 노화보건지소  34.177240  126.571151    노화읍  

[2612 rows x 8 columns]

In [32]:
# 결측치 확인
입지후보지.isnull().sum()

요양기관명    0
중분류      0
시군구      0
시도       0
주소       0
위도       0
경도       0
시군구2     0
dtype: int64

In [33]:
# 형식 매핑을 위한 딕셔너리 정의
map1 = {
    '서울특별시' : '서울',
    '부산광역시' : '부산',
    '대구광역시' : '대구',
    '인천광역시' : '인천',
    '광주광역시' : '광주',
    '대전광역시' : '대전',
    '울산광역시' : '울산',
    '세종특별자치시' : '세종특별자치시',
    '경기도' : '경기',
    '강원특별자치도' : '강원',
    '충청북도' : '충북',
    '충청남도' : '충남',
    '전북특별자치도' : '전북',
    '전라남도' : '전남',
    '경상북도' : '경북',
    '경상남도' : '경남',
    '제주특별자치도' : '제주'
}

In [34]:
# 매핑
입지후보지['시도2'] = 입지후보지['시도'].astype(str).map(map1)

In [35]:
입지후보지

요양기관명    중분류  시군구       시도  \
0         (사)인구보건복지협회 부산시지회 가족보건의원  21140  수영구    부산광역시   
1        (사)인구보건복지협회 충북세종지회 가족보건의원  33040  청주시     충청북도   
2             (사단법인)한국건강관리협회건강증진의원  24020   서구    광주광역시   
3               (사회복지법인)한국혈우재단광주의원  24020   서구    광주광역시   
4     (재)한국산업보건연구재단 서울DMC건강의원 포항분원  37010  포항시     경상북도   
...                            ...    ...  ...      ...   
2607                        강북구보건소  11090  강북구    서울특별시   
2608                    대구광역시서구보건소  22030   서구    대구광역시   
2609                        은평구보건소  11120  은평구    서울특별시   
2610                    군산시어청도보건지소      0  군산시  전북특별자치도   
2611                    완도군노화읍보건지소  36460  완도군     전라남도   

                                              주소         위도          경도  \
0                       부산광역시 수영구 수영로 425, (남천동)  35.144097  129.108936   
1            충청북도 청주시 흥덕구 내수동로 20, (복대동, 모자보건센타)  36.632997  127.449528   
2     광주광역시 서구 대남대로 432, 한국건강관리협회 지하2~지상7층 (농성동)  35.149037  126.888342   
3                     광주광역시 서구 죽봉대로 66, 4층 (농성동)  35.159100  126.883799   
4          경상북도 포항시 남구 오천읍 중원로 71-11, 1동 1층 101호  35.975330  129.400240   
...                                          ...        ...         ...   
2607      서울특별시 강북구 한천로 897, (번동, 강북보건소,번2동주민센터)  37.632333  127.038767   
2608                   대구광역시 서구 국채보상로 257, (평리동)  35.871815  128.559177   
2609                    서울특별시 은평구 은평로 195, (녹번동)  37.602671  126.928877   
2610              전북특별자치도 군산시 옥도면 어청도길 98, (옥도면)  36.120882  125.980074   
2611                전라남도 완도군 노화읍 노화로 813, 노화보건지소  34.177240  126.571151   

       시군구2 시도2  
0       수영로  부산  
1       흥덕구  충북  
2      대남대로  광주  
3      죽봉대로  광주  
4        남구  경북  
...     ...  ..  
2607    한천로  서울  
2608  국채보상로  대구  
2609    은평로  서울  
2610    옥도면  전북  
2611    노화읍  전남  

[2612 rows x 9 columns]

In [36]:
# 시도시군구 컬럼 생성 >> 시도시군구 코드를 추가할 때 기준으로 사용하기 위함
입지후보지['시도시군구'] = 입지후보지['시도2'] + ' ' + 입지후보지['시군구']
입지후보지

요양기관명    중분류  시군구       시도  \
0         (사)인구보건복지협회 부산시지회 가족보건의원  21140  수영구    부산광역시   
1        (사)인구보건복지협회 충북세종지회 가족보건의원  33040  청주시     충청북도   
2             (사단법인)한국건강관리협회건강증진의원  24020   서구    광주광역시   
3               (사회복지법인)한국혈우재단광주의원  24020   서구    광주광역시   
4     (재)한국산업보건연구재단 서울DMC건강의원 포항분원  37010  포항시     경상북도   
...                            ...    ...  ...      ...   
2607                        강북구보건소  11090  강북구    서울특별시   
2608                    대구광역시서구보건소  22030   서구    대구광역시   
2609                        은평구보건소  11120  은평구    서울특별시   
2610                    군산시어청도보건지소      0  군산시  전북특별자치도   
2611                    완도군노화읍보건지소  36460  완도군     전라남도   

                                              주소         위도          경도  \
0                       부산광역시 수영구 수영로 425, (남천동)  35.144097  129.108936   
1            충청북도 청주시 흥덕구 내수동로 20, (복대동, 모자보건센타)  36.632997  127.449528   
2     광주광역시 서구 대남대로 432, 한국건강관리협회 지하2~지상7층 (농성동)  35.149037  126.888342   
3                     광주광역시 서구 죽봉대로 66, 4층 (농성동)  35.159100  126.883799   
4          경상북도 포항시 남구 오천읍 중원로 71-11, 1동 1층 101호  35.975330  129.400240   
...                                          ...        ...         ...   
2607      서울특별시 강북구 한천로 897, (번동, 강북보건소,번2동주민센터)  37.632333  127.038767   
2608                   대구광역시 서구 국채보상로 257, (평리동)  35.871815  128.559177   
2609                    서울특별시 은평구 은평로 195, (녹번동)  37.602671  126.928877   
2610              전북특별자치도 군산시 옥도면 어청도길 98, (옥도면)  36.120882  125.980074   
2611                전라남도 완도군 노화읍 노화로 813, 노화보건지소  34.177240  126.571151   

       시군구2 시도2   시도시군구  
0       수영로  부산  부산 수영구  
1       흥덕구  충북  충북 청주시  
2      대남대로  광주   광주 서구  
3      죽봉대로  광주   광주 서구  
4        남구  경북  경북 포항시  
...     ...  ..     ...  
2607    한천로  서울  서울 강북구  
2608  국채보상로  대구   대구 서구  
2609    은평로  서울  서울 은평구  
2610    옥도면  전북  전북 군산시  
2611    노화읍  전남  전남 완도군  

[2612 rows x 10 columns]

# 한국행정구역분류 행정동코드 추출

In [37]:
path = '/content/drive/MyDrive/mulcam_bigdata/2024통계데이터공모전/기획_데이터수집/성원님/temp/한국행정구역분류_행정동코드(7자리)_20210701기준.xlsx'
dong = pd.read_excel(path, sheet_name='1. 총괄표(현행)', header=2, index_col=None)

In [39]:
dong = dong.drop(columns=[dong.columns[0]])
dong.fillna(0, inplace=True)
dong

시도      중분류   시군구        소분류  읍면동           영문 표기  한자 표기
0       서울특별시      0.0     0        0.0    0           Seoul  서울特別市
1       서울특별시  11010.0   종로구        0.0    0       Jongno-gu    鐘路區
2       서울특별시  11010.0   종로구  1101053.0  사직동      Sajik-dong    社稷洞
3       서울특별시  11010.0   종로구  1101054.0  삼청동  Samcheong-dong    三淸洞
4       서울특별시  11010.0   종로구  1101055.0  부암동       Buam-dong    付岩洞
...       ...      ...   ...        ...  ...             ...    ...
3787  제주특별자치도  39020.0  서귀포시  3902058.0  서홍동    Seohong-dong    西烘洞
3788  제주특별자치도  39020.0  서귀포시  3902059.0  대륜동    Daeryun-dong    大倫洞
3789  제주특별자치도  39020.0  서귀포시  3902060.0  대천동   Daecheon-dong    大川洞
3790  제주특별자치도  39020.0  서귀포시  3902061.0  중문동    Jungmun-dong    中文洞
3791  제주특별자치도  39020.0  서귀포시  3902062.0  예래동      Yerae-dong    猊來洞

[3792 rows x 7 columns]

In [40]:
# '중분류' 열을 문자열로 변환하면서 소수점 제거
dong['중분류'] = dong['중분류'].apply(lambda x: str(int(x)) if pd.notnull(x) else '')

# 결과 출력
dong

시도    중분류   시군구        소분류  읍면동           영문 표기  한자 표기
0       서울특별시      0     0        0.0    0           Seoul  서울特別市
1       서울특별시  11010   종로구        0.0    0       Jongno-gu    鐘路區
2       서울특별시  11010   종로구  1101053.0  사직동      Sajik-dong    社稷洞
3       서울특별시  11010   종로구  1101054.0  삼청동  Samcheong-dong    三淸洞
4       서울특별시  11010   종로구  1101055.0  부암동       Buam-dong    付岩洞
...       ...    ...   ...        ...  ...             ...    ...
3787  제주특별자치도  39020  서귀포시  3902058.0  서홍동    Seohong-dong    西烘洞
3788  제주특별자치도  39020  서귀포시  3902059.0  대륜동    Daeryun-dong    大倫洞
3789  제주특별자치도  39020  서귀포시  3902060.0  대천동   Daecheon-dong    大川洞
3790  제주특별자치도  39020  서귀포시  3902061.0  중문동    Jungmun-dong    中文洞
3791  제주특별자치도  39020  서귀포시  3902062.0  예래동      Yerae-dong    猊來洞

[3792 rows x 7 columns]

In [41]:
# 데이터 조작을 위한 원본 살리기
dong_temp = dong.copy()
dong_temp

시도    중분류   시군구        소분류  읍면동           영문 표기  한자 표기
0       서울특별시      0     0        0.0    0           Seoul  서울特別市
1       서울특별시  11010   종로구        0.0    0       Jongno-gu    鐘路區
2       서울특별시  11010   종로구  1101053.0  사직동      Sajik-dong    社稷洞
3       서울특별시  11010   종로구  1101054.0  삼청동  Samcheong-dong    三淸洞
4       서울특별시  11010   종로구  1101055.0  부암동       Buam-dong    付岩洞
...       ...    ...   ...        ...  ...             ...    ...
3787  제주특별자치도  39020  서귀포시  3902058.0  서홍동    Seohong-dong    西烘洞
3788  제주특별자치도  39020  서귀포시  3902059.0  대륜동    Daeryun-dong    大倫洞
3789  제주특별자치도  39020  서귀포시  3902060.0  대천동   Daecheon-dong    大川洞
3790  제주특별자치도  39020  서귀포시  3902061.0  중문동    Jungmun-dong    中文洞
3791  제주특별자치도  39020  서귀포시  3902062.0  예래동      Yerae-dong    猊來洞

[3792 rows x 7 columns]

In [43]:
# 데이터 조작에 불필요한 열 제거
dong_temp = dong_temp.drop(['소분류','영문 표기','한자 표기'], axis=1)
dong_temp

시도    중분류   시군구  읍면동
0       서울특별시      0     0    0
1       서울특별시  11010   종로구    0
2       서울특별시  11010   종로구  사직동
3       서울특별시  11010   종로구  삼청동
4       서울특별시  11010   종로구  부암동
...       ...    ...   ...  ...
3787  제주특별자치도  39020  서귀포시  서홍동
3788  제주특별자치도  39020  서귀포시  대륜동
3789  제주특별자치도  39020  서귀포시  대천동
3790  제주특별자치도  39020  서귀포시  중문동
3791  제주특별자치도  39020  서귀포시  예래동

[3792 rows x 4 columns]

In [44]:
# 형식 매핑을 위한 딕셔너리 정의
map3 = {
    '서울특별시' : '서울',
    '부산광역시' : '부산',
    '대구광역시' : '대구',
    '인천광역시' : '인천',
    '광주광역시' : '광주',
    '대전광역시' : '대전',
    '울산광역시' : '울산',
    '세종특별자치시' : '세종특별자치시',
    '경기도' : '경기',
    '강원도' : '강원',
    '충청북도' : '충북',
    '충청남도' : '충남',
    '전라북도' : '전북',
    '전라남도' : '전남',
    '경상북도' : '경북',
    '경상남도' : '경남',
    '제주특별자치도' : '제주'
}

In [45]:
# 매핑
dong_temp['시도2'] = dong_temp['시도'].astype(str).map(map3)
dong_temp

시도    중분류   시군구  읍면동 시도2
0       서울특별시      0     0    0  서울
1       서울특별시  11010   종로구    0  서울
2       서울특별시  11010   종로구  사직동  서울
3       서울특별시  11010   종로구  삼청동  서울
4       서울특별시  11010   종로구  부암동  서울
...       ...    ...   ...  ...  ..
3787  제주특별자치도  39020  서귀포시  서홍동  제주
3788  제주특별자치도  39020  서귀포시  대륜동  제주
3789  제주특별자치도  39020  서귀포시  대천동  제주
3790  제주특별자치도  39020  서귀포시  중문동  제주
3791  제주특별자치도  39020  서귀포시  예래동  제주

[3792 rows x 5 columns]

In [46]:
# 결측치 확인 후 처리
dong_temp[dong_temp['시도2'].isnull()]

시도 중분류 시군구 읍면동  시도2
452   0   0   0   0  NaN
675   0   0   0   0  NaN
826   0   0   0   0  NaN
993   0   0   0   0  NaN
1096  0   0   0   0  NaN
1182  0   0   0   0  NaN
1245  0   0   0   0  NaN
1268  0   0   0   0  NaN
1867  0   0   0   0  NaN
2080  0   0   0   0  NaN
2250  0   0   0   0  NaN
2476  0   0   0   0  NaN
2737  0   0   0   0  NaN
3058  0   0   0   0  NaN
3415  0   0   0   0  NaN
3745  0   0   0   0  NaN

In [47]:
# 시도2 열의 결측치가진 행 삭제 dropna
dong_temp = dong_temp.dropna(subset=['시도2'])

In [48]:
# 결측치 삭제 결과 확인
dong_temp['시도2'].isnull().sum()

0

In [49]:
# 0인 행 제외한 df로 업데이트
dong_temp = dong_temp[dong_temp['시군구'] != 0]

In [50]:
dong_temp['시군구'].unique()

array(['종로구', '중구', '용산구', '성동구', '광진구', '동대문구', '중랑구', '성북구', '강북구',
       '도봉구', '노원구', '은평구', '서대문구', '마포구', '양천구', '강서구', '구로구', '금천구',
       '영등포구', '동작구', '관악구', '서초구', '강남구', '송파구', '강동구', '서구', '동구',
       '영도구', '부산진구', '동래구', '남구', '북구', '해운대구', '사하구', '금정구', '연제구',
       '수영구', '사상구', '기장군', '수성구', '달서구', '달성군', '연수구', '남동구', '부평구',
       '계양구', '미추홀구', '강화군', '옹진군', '광산구', '유성구', '대덕구', '울주군', '세종특별자치시',
       '수원시', '장안구', '권선구', '팔달구', '영통구', '성남시', '수정구', '중원구', '분당구',
       '의정부시', '안양시', '만안구', '동안구', '부천시', '광명시', '평택시', '동두천시', '안산시',
       '상록구', '단원구', '고양시', '덕양구', '일산동구', '일산서구', '과천시', '구리시', '남양주시',
       '오산시', '시흥시', '군포시', '의왕시', '하남시', '용인시', '처인구', '기흥구', '수지구',
       '파주시', '이천시', '안성시', '김포시', '화성시', '광주시', '양주시', '포천시', '여주시',
       '연천군', '가평군', '양평군', '춘천시', '원주시', '강릉시', '동해시', '태백시', '속초시',
       '삼척시', '홍천군', '횡성군', '영월군', '평창군', '정선군', '철원군', '화천군', '양구군',
       '인제군', '고성군', '양양군', '충주시', '제천시', '청주시', '상당구', '서원구', '흥덕구',
       '청원

In [51]:
dong_temp['시도시군구'] = dong_temp['시도2'] + ' ' + dong_temp['시군구']
dong_temp

시도    중분류   시군구  읍면동 시도2    시도시군구
1       서울특별시  11010   종로구    0  서울   서울 종로구
2       서울특별시  11010   종로구  사직동  서울   서울 종로구
3       서울특별시  11010   종로구  삼청동  서울   서울 종로구
4       서울특별시  11010   종로구  부암동  서울   서울 종로구
5       서울특별시  11010   종로구  평창동  서울   서울 종로구
...       ...    ...   ...  ...  ..      ...
3787  제주특별자치도  39020  서귀포시  서홍동  제주  제주 서귀포시
3788  제주특별자치도  39020  서귀포시  대륜동  제주  제주 서귀포시
3789  제주특별자치도  39020  서귀포시  대천동  제주  제주 서귀포시
3790  제주특별자치도  39020  서귀포시  중문동  제주  제주 서귀포시
3791  제주특별자치도  39020  서귀포시  예래동  제주  제주 서귀포시

[3759 rows x 6 columns]

In [52]:
dong_temp_res = pd.DataFrame(dong_temp.groupby(['중분류', '시도시군구'])['읍면동'].count())
dong_temp_res

읍면동
중분류   시도시군구       
11010 서울 종로구    18
11020 서울 중구     16
11030 서울 용산구    17
11040 서울 성동구    18
11050 서울 광진구    16
...            ...
38380 경남 함양군    12
38390 경남 거창군    13
38400 경남 합천군    18
39010 제주 제주시    27
39020 제주 서귀포시   18

[261 rows x 1 columns]

In [53]:
dong_temp_res = dong_temp_res.reset_index()
dong_temp_res = dong_temp_res.drop('읍면동', axis=1)
dong_temp_res

중분류    시도시군구
0    11010   서울 종로구
1    11020    서울 중구
2    11030   서울 용산구
3    11040   서울 성동구
4    11050   서울 광진구
..     ...      ...
256  38380   경남 함양군
257  38390   경남 거창군
258  38400   경남 합천군
259  39010   제주 제주시
260  39020  제주 서귀포시

[261 rows x 2 columns]

In [54]:
%pwd

'/content'

In [57]:
# dong_temp_res.to_csv('/content/drive/MyDrive/mulcam_bigdata/2024통계데이터공모전/기획_데이터수집/성원님/temp/행정구역분류_행정동코드_정리.csv', index=False)

# 두 데이터 병합

In [59]:
code = pd.read_csv('/content/drive/MyDrive/mulcam_bigdata/2024통계데이터공모전/기획_데이터수집/성원님/temp/행정구역분류_행정동코드_정리.csv')

In [63]:
df_hos = 입지후보지.drop('중분류', axis=1)
df_hos

요양기관명  시군구       시도  \
0         (사)인구보건복지협회 부산시지회 가족보건의원  수영구    부산광역시   
1        (사)인구보건복지협회 충북세종지회 가족보건의원  청주시     충청북도   
2             (사단법인)한국건강관리협회건강증진의원   서구    광주광역시   
3               (사회복지법인)한국혈우재단광주의원   서구    광주광역시   
4     (재)한국산업보건연구재단 서울DMC건강의원 포항분원  포항시     경상북도   
...                            ...  ...      ...   
2607                        강북구보건소  강북구    서울특별시   
2608                    대구광역시서구보건소   서구    대구광역시   
2609                        은평구보건소  은평구    서울특별시   
2610                    군산시어청도보건지소  군산시  전북특별자치도   
2611                    완도군노화읍보건지소  완도군     전라남도   

                                              주소         위도          경도  \
0                       부산광역시 수영구 수영로 425, (남천동)  35.144097  129.108936   
1            충청북도 청주시 흥덕구 내수동로 20, (복대동, 모자보건센타)  36.632997  127.449528   
2     광주광역시 서구 대남대로 432, 한국건강관리협회 지하2~지상7층 (농성동)  35.149037  126.888342   
3                     광주광역시 서구 죽봉대로 66, 4층 (농성동)  35.159100  126.883799   
4          경상북도 포항시 남구 오천읍 중원로 71-11, 1동 1층 101호  35.975330  129.400240   
...                                          ...        ...         ...   
2607      서울특별시 강북구 한천로 897, (번동, 강북보건소,번2동주민센터)  37.632333  127.038767   
2608                   대구광역시 서구 국채보상로 257, (평리동)  35.871815  128.559177   
2609                    서울특별시 은평구 은평로 195, (녹번동)  37.602671  126.928877   
2610              전북특별자치도 군산시 옥도면 어청도길 98, (옥도면)  36.120882  125.980074   
2611                전라남도 완도군 노화읍 노화로 813, 노화보건지소  34.177240  126.571151   

       시군구2 시도2   시도시군구  
0       수영로  부산  부산 수영구  
1       흥덕구  충북  충북 청주시  
2      대남대로  광주   광주 서구  
3      죽봉대로  광주   광주 서구  
4        남구  경북  경북 포항시  
...     ...  ..     ...  
2607    한천로  서울  서울 강북구  
2608  국채보상로  대구   대구 서구  
2609    은평로  서울  서울 은평구  
2610    옥도면  전북  전북 군산시  
2611    노화읍  전남  전남 완도군  

[2612 rows x 9 columns]

In [61]:
code

중분류    시도시군구
0    11010   서울 종로구
1    11020    서울 중구
2    11030   서울 용산구
3    11040   서울 성동구
4    11050   서울 광진구
..     ...      ...
256  38380   경남 함양군
257  38390   경남 거창군
258  38400   경남 합천군
259  39010   제주 제주시
260  39020  제주 서귀포시

[261 rows x 2 columns]

In [64]:
result_temp = df_hos.merge(code, on='시도시군구', how='left', suffixes=('_x','_y'))
result_temp

요양기관명  시군구       시도  \
0         (사)인구보건복지협회 부산시지회 가족보건의원  수영구    부산광역시   
1        (사)인구보건복지협회 충북세종지회 가족보건의원  청주시     충청북도   
2             (사단법인)한국건강관리협회건강증진의원   서구    광주광역시   
3               (사회복지법인)한국혈우재단광주의원   서구    광주광역시   
4     (재)한국산업보건연구재단 서울DMC건강의원 포항분원  포항시     경상북도   
...                            ...  ...      ...   
2607                        강북구보건소  강북구    서울특별시   
2608                    대구광역시서구보건소   서구    대구광역시   
2609                        은평구보건소  은평구    서울특별시   
2610                    군산시어청도보건지소  군산시  전북특별자치도   
2611                    완도군노화읍보건지소  완도군     전라남도   

                                              주소         위도          경도  \
0                       부산광역시 수영구 수영로 425, (남천동)  35.144097  129.108936   
1            충청북도 청주시 흥덕구 내수동로 20, (복대동, 모자보건센타)  36.632997  127.449528   
2     광주광역시 서구 대남대로 432, 한국건강관리협회 지하2~지상7층 (농성동)  35.149037  126.888342   
3                     광주광역시 서구 죽봉대로 66, 4층 (농성동)  35.159100  126.883799   
4          경상북도 포항시 남구 오천읍 중원로 71-11, 1동 1층 101호  35.975330  129.400240   
...                                          ...        ...         ...   
2607      서울특별시 강북구 한천로 897, (번동, 강북보건소,번2동주민센터)  37.632333  127.038767   
2608                   대구광역시 서구 국채보상로 257, (평리동)  35.871815  128.559177   
2609                    서울특별시 은평구 은평로 195, (녹번동)  37.602671  126.928877   
2610              전북특별자치도 군산시 옥도면 어청도길 98, (옥도면)  36.120882  125.980074   
2611                전라남도 완도군 노화읍 노화로 813, 노화보건지소  34.177240  126.571151   

       시군구2 시도2   시도시군구      중분류  
0       수영로  부산  부산 수영구  21140.0  
1       흥덕구  충북  충북 청주시  33040.0  
2      대남대로  광주   광주 서구  24020.0  
3      죽봉대로  광주   광주 서구  24020.0  
4        남구  경북  경북 포항시  37010.0  
...     ...  ..     ...      ...  
2607    한천로  서울  서울 강북구  11090.0  
2608  국채보상로  대구   대구 서구  22030.0  
2609    은평로  서울  서울 은평구  11120.0  
2610    옥도면  전북  전북 군산시  35020.0  
2611    노화읍  전남  전남 완도군  36460.0  

[2612 rows x 10 columns]

In [72]:
# 결측치 : 세종 (29010)
result_temp.loc[result_temp['시도'] == '세종특별자치시', '중분류'] = 29010.0

In [73]:
result_temp[result_temp['시도'] == '세종특별자치시']

요양기관명    시군구       시도  \
102    고운맘소아청소년과의원    마음로  세종특별자치시   
105     고운소아청소년과의원   남세종로  세종특별자치시   
126   곰돌이 소아청소년과의원    달빛로  세종특별자치시   
356     꿈담소아청소년과의원    마음로  세종특별자치시   
441     다정소아청소년과의원   다정북로  세종특별자치시   
478   도담도담소아청소년과의원   보듬3로  세종특별자치시   
491   도란도란소아청소년과의원  다정중앙로  세종특별자치시   
548     라온소아청소년과의원  새롬중앙로  세종특별자치시   
564     로뎀소아청소년과의원  다정중앙로  세종특별자치시   
645    무지개소아청소년과의원   소담1로  세종특별자치시   
1126    세움소아청소년과의원   보듬3로  세종특별자치시   
1131  세종성모소아청소년과의원  한누리대로  세종특별자치시   
1133    세종소아청소년과의원  한누리대로  세종특별자치시   
1135  세종아이소아청소년과의원    달빛로  세종특별자치시   
1136  세종연세소아청소년과의원  새롬중앙로  세종특별자치시   
1141  소담튼튼소아청소년과의원  한누리대로  세종특별자치시   
1153   손근형소아청소년과의원   나성북로  세종특별자치시   
1254  아름누리소아청소년과의원  한누리대로  세종특별자치시   
1767   웰키즈소아청소년과의원   보듬3로  세종특별자치시   
1799     유소아청소년과의원   도움1로  세종특별자치시   
1813    윤빛소아청소년과의원   해밀3로  세종특별자치시   
1880    이삭소아청소년과의원  한누리대로  세종특별자치시   
1919   이은경소아청소년과의원   조치원읍  세종특별자치시   
2158    중앙소아청소년과의원  한누리대로  세종특별자치시   
2272   코끼리소아청소년과의원  한누리대로  세종특별자치시   
2274    코난소아청소년과의원   남세종로  세종특별자치시   
2430    하늘소아청소년과의원  한누리대로  세종특별자치시   

                                                     주소         위도  \
102                   세종특별자치시 마음로 70, 304호 (고운동, 에셀프라자)  36.505074   
105                세종특별자치시 남세종로 454, 405호 (보람동, 강남제일타워)  36.477293   
126                 세종특별자치시 달빛로 47, 302호 (종촌동, 종촌파크프라자)  36.503154   
356              세종특별자치시 마음로 264, 고운프라자 2층 201-1호 (고운동)  36.519279   
441           세종특별자치시 다정북로 109, 204,205호 (다정동, 가온마을3단지)  36.497353   
478             세종특별자치시 보듬3로 6, 212호 (도담동, 라온프라이빗시티2상가)  36.512824   
491                세종특별자치시 다정중앙로 57, 다정센타프라자 303호 (다정동)  36.495624   
548           세종특별자치시 새롬중앙로 62-15, 해피라움W 303,304호 (새롬동)  36.486911   
564              세종특별자치시 다정중앙로 52, 페스티발2 402~404호 (다정동)  36.495114   
645                 세종특별자치시 소담1로 16, 406호 (소담동, 지엘플렉스2)  36.484553   
1126               세종특별자치시 보듬3로 92, 310호 (아름동, 해피라움2상가)  36.511737   
1131  세종특별자치시 한누리대로 2129, 스타힐타워Ⅱ 401~405,409,412~415...  36.478300   
1133               세종특별자치시 한누리대로 157, 405호 (나성동, 금강프라자)  36.479153   
1135               세종특별자치시 달빛로 59, 303.304호 (종촌동, 호만빌딩)  36.504215   
1136             세종특별자치시 새롬중앙로 47, 402~403호 (새롬동, 세움빌딩)  36.484881   
1141          세종특별자치시 한누리대로 2016, 404호,405호 (소담동, 타임빌딩)  36.485076   
1153        세종특별자치시 나성북로 30, 퍼스트원 5층 563,564,565호 (나성동)  36.490171   
1254         세종특별자치시 한누리대로 2022, 아이콤스타타워 402,403호 (소담동)  36.484494   
1767  세종특별자치시 보듬3로 95, 118,308,310,401~5,410,603~8,7...  36.511367   
1799              세종특별자치시 도움1로 108, 408.409호 (종촌동, 몰리브)  36.502649   
1813                  세종특별자치시 해밀3로 43, 스카이드림 303호 (해밀동)  36.528483   
1880       세종특별자치시 한누리대로 169, 세종행복타워 301,307,401호 (나성동)  36.480108   
1919                        세종특별자치시 조치원읍 새내로 77, (조치원읍)  36.598668   
2158          세종특별자치시 한누리대로 2257, 판타지타워 404, 405호 (대평동)  36.472680   
2272          세종특별자치시 한누리대로 2012, 코끼리타워 401호,402호 (소담동)  36.485374   
2274                 세종특별자치시 남세종로 73, 1층 109,110호 (집현동)  36.480141   
2430  세종특별자치시 한누리대로 1820, 왕관빌딩 314-2,315,404,405-1호 ...  36.494439   

              경도    시군구2      시도2          시도시군구      중분류  
102   127.234400     70,  세종특별자치시    세종특별자치시 마음로  29010.0  
105   127.289610    454,  세종특별자치시   세종특별자치시 남세종로  29010.0  
126   127.249892     47,  세종특별자치시    세종특별자치시 달빛로  29010.0  
356   127.234660    264,  세종특별자치시    세종특별자치시 마음로  29010.0  
441   127.242584    109,  세종특별자치시   세종특별자치시 다정북로  29010.0  
478   127.258865      6,  세종특별자치시   세종특별자치시 보듬3로  29010.0  
491   127.248519     57,  세종특별자치시  세종특별자치시 다정중앙로  29010.0  
548   127.251168  62-15,  세종특별자치시  세종특별자치시 새롬중앙로  29010.0  
564   127.249274     52,  세종특별자치시  세종특별자치시 다정중앙로  29010.0  
645   127.301110     16,  세종특별자치시   세종특별자치시 소담1로  29010.0  
1126  127.249728     92,  세종특별자치시   세종특별자치시 보듬3로  29010.0  
1131  127.289300   2129,  세종특별자치시  세종특별자치시 한누리대로  29010.0  
1133  127.262401    157,  세종특별자치시  세종특별자치시 한누리대로  29010.0  
1135  127.249758     59,  세종특별자치시    세종특별자치시 달빛로  29010.0  
1136  127.250796     47,  세종특별자치시  세종특별자치시 새롬중앙로  29010.0  
1141  127.299616   2016,  세

In [74]:
result_temp.isnull().sum()

요양기관명    0
시군구      0
시도       0
주소       0
위도       0
경도       0
시군구2     0
시도2      0
시도시군구    0
중분류      0
dtype: int64

In [78]:
result_df = pd.DataFrame(result_temp.groupby('중분류')['시군구'].count()).reset_index()
result_df

중분류  시군구
0    11010.0    6
1    11020.0    7
2    11030.0   14
3    11040.0   20
4    11050.0   18
..       ...  ...
174  38350.0    1
175  38380.0    1
176  38390.0    2
177  39010.0   32
178  39020.0    8

[179 rows x 2 columns]

In [84]:
result_df['중분류'] = result_df['중분류'].astype(int)

In [85]:
result_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 179 entries, 0 to 178
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   중분류     179 non-null    int64
 1   시군구     179 non-null    int64
dtypes: int64(2)
memory usage: 2.9 KB


In [79]:
result_df['시군구'].sum()

2612

In [81]:
path2 = '/content/drive/MyDrive/mulcam_bigdata/2024통계데이터공모전/기획_데이터수집/성원님/temp/시군구코드별_변수_v3.csv'
df = pd.read_csv(path2)
df

중분류 행정구역시도명 행정구역시군구명      면적  소아청소년수  출생아수  가임연령부부수  2차병원수  3차병원수  \
0    31050     경기도      부천시   53.45    2227   791     1203    8.0    0.0   
1    31240     경기도      화성시  698.18    2319   991     1450   10.0    0.0   
2    31023     경기도      분당구   69.49    2093   580      969    5.0    1.0   
3    31130     경기도     남양주시  458.14    1816   646      827    5.0    0.0   
4    11230   서울특별시      강남구   39.50    1634   369      576    7.0    0.0   
..     ...     ...      ...     ...     ...   ...      ...    ...    ...   
245  32410     강원도      양양군  630.01      99    42       53    0.0    0.0   
246  37330    경상북도      청송군  846.12      94    39       46    1.0    0.0   
247  37340    경상북도      영양군  815.75      78    24       24    0.0    0.0   
248  37310    경상북도      군위군  614.32      69    21       25    0.0    0.0   
249  37430    경상북도      울릉군   72.94      51    22       30    1.0    0.0   

     달빛병원수  
0      1.0  
1      2.0  
2      0.0  
3      1.0  
4      2.0  
..     ...  
245    0.0  
246    0.0  
247    0.0  
248    0.0  
249    0.0  

[250 rows x 10 columns]

In [82]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 250 entries, 0 to 249
Data columns (total 10 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   중분류       250 non-null    int64  
 1   행정구역시도명   250 non-null    object 
 2   행정구역시군구명  250 non-null    object 
 3   면적        250 non-null    float64
 4   소아청소년수    250 non-null    int64  
 5   출생아수      250 non-null    int64  
 6   가임연령부부수   250 non-null    int64  
 7   2차병원수     250 non-null    float64
 8   3차병원수     250 non-null    float64
 9   달빛병원수     250 non-null    float64
dtypes: float64(4), int64(4), object(2)
memory usage: 19.7+ KB


In [92]:
df['중분류'].values

array([31050, 31240, 31023, 31130, 11230, 22070, 31192, 31193, 23080,
       22050, 21090, 23060, 11240, 23040, 23050, 38070, 31150, 11110,
       31200, 22060, 31230, 11220, 24050, 29010, 39010, 31070, 11150,
       31014, 24040, 34012, 11160, 25030, 34040, 31101, 31104, 31030,
       25040, 35012, 37050, 38100, 31060, 31042, 23070, 11250, 31092,
       26020, 24020, 31103, 11080, 35011, 31140, 38090, 11120, 23090,
       31250, 38030, 31012, 38111, 31091, 32020, 26040, 11170, 11070,
       38115, 38112, 26030, 33044, 33043, 36010, 34011, 31011, 11190,
       26010, 21060, 24030, 37100, 31021, 11200, 21070, 31160, 37012,
       22310, 35030, 22020, 35020, 31180, 21100, 21310, 11130, 11210,
       36030, 11140, 31191, 37011, 31260, 11050, 11060, 21130, 26310,
       33041, 32010, 38114, 34080, 31120, 25050, 11100, 31110, 21050,
       31210, 21080, 33042, 31022, 36020, 31041, 25020, 25010, 31170,
       36060, 34070, 23010, 34050, 38113, 11090, 21110, 31220, 11030,
       11010, 21150,

In [91]:
result_df['중분류'].values

array([11010, 11020, 11030, 11040, 11050, 11060, 11070, 11080, 11090,
       11100, 11110, 11120, 11130, 11140, 11150, 11160, 11170, 11180,
       11190, 11200, 11210, 11220, 11230, 11240, 11250, 21010, 21020,
       21030, 21040, 21050, 21060, 21070, 21080, 21090, 21100, 21110,
       21120, 21130, 21140, 21150, 21310, 22010, 22020, 22030, 22040,
       22050, 22060, 22070, 22310, 23010, 23020, 23040, 23050, 23060,
       23070, 23080, 23090, 23310, 24010, 24020, 24030, 24040, 24050,
       25010, 25020, 25030, 25040, 25050, 26010, 26020, 26030, 26040,
       26310, 29010, 31010, 31020, 31030, 31040, 31050, 31060, 31070,
       31080, 31090, 31100, 31110, 31120, 31130, 31140, 31150, 31160,
       31170, 31180, 31190, 31200, 31210, 31220, 31230, 31240, 31250,
       31260, 31270, 31280, 31380, 32010, 32020, 32030, 32040, 32050,
       32060, 32070, 32310, 32320, 32380, 33020, 33030, 33040, 33330,
       33340, 33350, 33370, 33390, 34010, 34020, 34030, 34040, 34050,
       34060, 34070,

1차병원수 2612개인데 우리 데이터에는 2112개만 묶여서 누락된 데이터 원인 확인
- 우리 데이터에 없는 시도별코드를 가진 병원수 >> 500개
- 따라서 우리 데이터 기준 250개 시도에는 오류 없이 잘 할당되었다

In [93]:
ok = []
no = []

for i in result_df['중분류'].values:
    if i in df['중분류'].values:
        ok.append(i)
    else:
        no.append(i)

In [94]:
no

[31010, 31020, 31040, 31090, 31100, 31190, 33040, 34010, 35010, 37010, 38110]

In [96]:
code[code['중분류'] == 31010]

중분류   시도시군구
75  31010  경기 수원시

In [98]:
df[df['중분류'] == 31010]

Empty DataFrame
Columns: [중분류, 행정구역시도명, 행정구역시군구명, 면적, 소아청소년수, 출생아수, 가임연령부부수, 2차병원수, 3차병원수, 달빛병원수]
Index: []

In [110]:
col = [31010, 31020, 31040, 31090, 31100, 31190, 33040, 34010, 35010, 37010, 38110]

74   NaN
Name: 시군구, dtype: float64

In [122]:
print(df[df['중분류'] == 31010])
print(df[df['중분류'] == 31020])
print(df[df['중분류'] == 31040])
print(df[df['중분류'] == 31090])
print(df[df['중분류'] == 31100])
print(df[df['중분류'] == 31190])
print(df[df['중분류'] == 33040])
print(df[df['중분류'] == 34010])
print(df[df['중분류'] == 35010])
print(df[df['중분류'] == 37010])
print(df[df['중분류'] == 38110])

Empty DataFrame
Columns: [중분류, 행정구역시도명, 행정구역시군구명, 면적, 소아청소년수, 출생아수, 가임연령부부수, 2차병원수, 3차병원수, 달빛병원수]
Index: []
Empty DataFrame
Columns: [중분류, 행정구역시도명, 행정구역시군구명, 면적, 소아청소년수, 출생아수, 가임연령부부수, 2차병원수, 3차병원수, 달빛병원수]
Index: []
Empty DataFrame
Columns: [중분류, 행정구역시도명, 행정구역시군구명, 면적, 소아청소년수, 출생아수, 가임연령부부수, 2차병원수, 3차병원수, 달빛병원수]
Index: []
Empty DataFrame
Columns: [중분류, 행정구역시도명, 행정구역시군구명, 면적, 소아청소년수, 출생아수, 가임연령부부수, 2차병원수, 3차병원수, 달빛병원수]
Index: []
Empty DataFrame
Columns: [중분류, 행정구역시도명, 행정구역시군구명, 면적, 소아청소년수, 출생아수, 가임연령부부수, 2차병원수, 3차병원수, 달빛병원수]
Index: []
Empty DataFrame
Columns: [중분류, 행정구역시도명, 행정구역시군구명, 면적, 소아청소년수, 출생아수, 가임연령부부수, 2차병원수, 3차병원수, 달빛병원수]
Index: []
Empty DataFrame
Columns: [중분류, 행정구역시도명, 행정구역시군구명, 면적, 소아청소년수, 출생아수, 가임연령부부수, 2차병원수, 3차병원수, 달빛병원수]
Index: []
Empty DataFrame
Columns: [중분류, 행정구역시도명, 행정구역시군구명, 면적, 소아청소년수, 출생아수, 가임연령부부수, 2차병원수, 3차병원수, 달빛병원수]
Index: []
Empty DataFrame
Columns: [중분류, 행정구역시도명, 행정구역시군구명, 면적, 소아청소년수, 출생아수, 가임연령부부수, 2차병원수, 3차병원수, 달빛병원수]
Index: []
Empty DataFrame
Columns: [중분

In [121]:
76+74+32+30+65+65+39+27+37+21+34

500

In [ ]:
print(result_df[result_df['중분류'] == 31010])
print(result_df[result_df['중분류'] == 31020])
print(result_df[result_df['중분류'] == 31040])
print(result_df[result_df['중분류'] == 31090])
print(result_df[result_df['중분류'] == 31100])
print(result_df[result_df['중분류'] == 31190])
print(result_df[result_df['중분류'] == 33040])
print(result_df[result_df['중분류'] == 34010])
print(result_df[result_df['중분류'] == 35010])
print(result_df[result_df['중분류'] == 37010])
print(result_df[result_df['중분류'] == 38110])

In [86]:
res = df.merge(result_df, on='중분류', how = 'left')
res

중분류 행정구역시도명 행정구역시군구명      면적  소아청소년수  출생아수  가임연령부부수  2차병원수  3차병원수  \
0    31050     경기도      부천시   53.45    2227   791     1203    8.0    0.0   
1    31240     경기도      화성시  698.18    2319   991     1450   10.0    0.0   
2    31023     경기도      분당구   69.49    2093   580      969    5.0    1.0   
3    31130     경기도     남양주시  458.14    1816   646      827    5.0    0.0   
4    11230   서울특별시      강남구   39.50    1634   369      576    7.0    0.0   
..     ...     ...      ...     ...     ...   ...      ...    ...    ...   
245  32410     강원도      양양군  630.01      99    42       53    0.0    0.0   
246  37330    경상북도      청송군  846.12      94    39       46    1.0    0.0   
247  37340    경상북도      영양군  815.75      78    24       24    0.0    0.0   
248  37310    경상북도      군위군  614.32      69    21       25    0.0    0.0   
249  37430    경상북도      울릉군   72.94      51    22       30    1.0    0.0   

     달빛병원수   시군구  
0      1.0  42.0  
1      2.0  62.0  
2      0.0   NaN  
3      1.0  46.0  
4      2.0  52.0  
..     ...   ...  
245    0.0   NaN  
246    0.0   NaN  
247    0.0   NaN  
248    0.0   NaN  
249    0.0   NaN  

[250 rows x 11 columns]

In [88]:
res['시군구'].sum()

2112.0

In [118]:
res.columns = ['중분류', '행정구역시도명', '행정구역시군구명', '면적', '소아청소년수', '출생아수', '가임연령부부수',
       '2차병원수', '3차병원수', '달빛병원수', '1차병원수']

In [124]:
res['1차병원수'] = res['1차병원수'].fillna(0)

In [126]:
# res.to_csv('/content/drive/MyDrive/mulcam_bigdata/2024통계데이터공모전/기획_데이터수집/성원님/temp/시군구코드별_변수_v4.csv', index=0)